### ISSUE TITLE
acfun某些视频解析错误

### ISSUE THREAD

**Issue by wj9559** [soimort/you-get#1696]
Please make sure these boxes are checked before submitting your issue – thank you!

- [ x ] You can actually watch the video in your browser or mobile application, but not download them with `you-get`.
- [ x ] Your `you-get` is up-to-date.
- [ x ] I have read <https://github.com/soimort/you-get/wiki/FAQ> and tried to do so.
- [ x ] The issue is not yet reported on <https://github.com/soimort/you-get/issues> or  <https://github.com/soimort/you-get/wiki/Known-Bugs>. If so, please add your comments under the existing issue.
- [ x ] The issue (or question) is really about `you-get`, not about some other code or project.

Run the command with the `--debug` option, and paste the full output inside the fences:

```
$ you-get --debug http://www.acfun.cn/v/ac3483593
[DEBUG] get_html: http://www.acfun.cn/v/ac3483593
[DEBUG] get_response: http://www.acfun.cn/v/ac3483593
[DEBUG] get_html: http://www.acfun.tv/video/getVideo.aspx?id=4880535
[DEBUG] get_response: http://www.acfun.tv/video/getVideo.aspx?id=4880535
[DEBUG] get_content: http://api.aixifan.com/plays/4880535
[DEBUG] get_content: https://api.youku.com/players/custom.json?client_id=908a519d032263f8&video_id=CMTE2MjM5NDA=&embsig=1_1487493522_f6321faede750e9c3874ddf3863654a3
[DEBUG] get_content: http://play.youku.com/partner/get.json?cid=908a519d032263f8&vid=CMTE2MjM5NDA=&ct=85&sign=MXwxNDg3NDkzNjMxfDc2ZTlmZTE4ODdlNjliMzgxNzMxMDc0Y2E1ZTJlY2Uw
[DEBUG] get_content: http://play.youku.com/partner/get.json?cid=908a519d032263f8&vid=CMTE2MjM5NDA=&ct=86&sign=MXwxNDg3NDkzNjMxfDc2ZTlmZTE4ODdlNjliMzgxNzMxMDc0Y2E1ZTJlY2Uw
you-get: [Failed] Video not found.
```

If there's anything else you would like to say (e.g. in case your issue is not about downloading a specific video; it might as well be a general discussion or proposal for a new feature), fill in the box below; otherwise, you may want to post an emoji or meme instead:

> [WRITE SOMETHING]
Only a small part of the video.
> [OR HAVE SOME :icecream:!]

**Comment by zhangn1985** [comment]
几乎是全部视频都无法下载
https://github.com/zhangn1985/ykdl/issues/61

**Comment by denobisipsis** [comment]
self.class.trans_e(
"328f45d8",
base64.b64decode(bytes(self.data, 'ascii'))
)

**Comment by zhangn1985** [comment]
the algorithm provided by @denobisipsis is right.
I have update my fork: https://github.com/zhangn1985/ykdl/commit/77aa1f980b6866fc611aca245989978dd52a64e0

**Comment by lytmkai** [comment]
`you-get --debug -o  E:\       			http://www.acfun.cn/v/ac3488466

[DEBUG] get_content: https://api.youku.com/players/custom.json?client_id=908a519d032263f8&video_id=58ac111d0cf218e4a4d0a050&embsig=1_1487688363_cd4f17ece5bbe03d97d93e885d386af7
[DEBUG] get_content: http://play.youku.com/partner/get.json?cid=908a519d032263f8&vid=58ac111d0cf218e4a4d0a050&ct=85&sign=MXwxNDg3Njg4MzgzfDI5MjE5YmE4ZTJlOGM2NDg5NzIxZDg4YjAzYTkwOTQ0
[DEBUG] get_content: http://play.youku.com/partner/get.json?cid=908a519d032263f8&vid=58ac111d0cf218e4a4d0a050&ct=86&sign=MXwxNDg3Njg4MzgzfDI5MjE5YmE4ZTJlOGM2NDg5NzIxZDg4YjAzYTkwOTQ0

you-get: [Failed] vid参数错误`


vid is 58ac111d0cf218e4a4d0a050, not 'CMTE2MjM5NDA=' format.

**Comment by freeleefly** [comment]
`>you-get --version
you-get: version 0.4.652, a tiny downloader that scrapes the web.`

`>you-get http://www.acfun.cn/v/ac3498310
you-get: [Failed] vid参数错误`

A站几乎全部参数错误

**Comment by wj9559** [comment]
weibo.com的视频无法解析,用 --debug 参数也完全没有输出, zhangn1985 fork 的[ykdl](https://github.com/zhangn1985/ykdl)可以
tudou.com的视频是卡了很久都没输出,加了 --debug 后是这样：
```
~$ you-get --debug "http://www.tudou.com/programs/view/3nB9zN0Y-Mw/?spm=a2h0k.8191414.3nB9zN0Y-Mw.A"
[DEBUG] get_decoded_html: http://www.tudou.com/programs/view/3nB9zN0Y-Mw/?spm=a2h0k.8191414.3nB9zN0Y-Mw.A
[DEBUG] get_response: http://www.tudou.com/programs/view/3nB9zN0Y-Mw/?spm=a2h0k.8191414.3nB9zN0Y-Mw.A
[DEBUG] get_decoded_html: http://www.tudou.com/outplay/goto/getItemSegs.action?iid=70286645
[DEBUG] get_response: http://www.tudou.com/outplay/goto/getItemSegs.action?iid=70286645
[DEBUG] get_html: http://ct.v2.tudou.com/f?id=41526820
[DEBUG] get_response: http://ct.v2.tudou.com/f?id=41526820
you-get: version 0.4.652, a tiny downloader that scrapes the web.
you-get: ['http://www.tudou.com/programs/view/3nB9zN0Y-Mw/?spm=a2h0k.8191414.3nB9zN0Y-Mw.A']
Traceback (most recent call last):
  File "/usr/bin/you-get", line 11, in <module>
    load_entry_point('you-get==0.4.652', 'console_scripts', 'you-get')()
  File "/usr/lib/python3.6/site-packages/you_get/__main__.py", line 92, in main
    main(**kwargs)
  File "/usr/lib/python3.6/site-packages/you_get/common.py", line 1421, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "/usr/lib/python3.6/site-packages/you_get/common.py", line 1334, in script_main
    download_main(download, download_playlist, args, playlist, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
  File "/usr/lib/python3.6/site-packages/you_get/common.py", line 1150, in download_main
    download(url, **kwargs)
  File "/usr/lib/python3.6/site-packages/you_get/common.py", line 1414, in any_download
    m.download(url, **kwargs)
  File "/usr/lib/python3.6/site-packages/you_get/extractors/tudou.py", line 60, in tudou_download
    tudou_download_by_iid(iid, title, output_dir = output_dir, merge = merge, info_only = info_only)
  File "/usr/lib/python3.6/site-packages/you_get/extractors/tudou.py", line 16, in tudou_download_by_iid
    for i in parseString(get_html('http://ct.v2.tudou.com/f?id=%s' % vid)).getElementsByTagName('f'):
  File "/usr/lib/python3.6/site-packages/you_get/common.py", line 282, in get_html
    content = get_response(url, faker).data
  File "/usr/lib/python3.6/site-packages/you_get/common.py", line 268, in get_response
    response = request.urlopen(url)
  File "/usr/lib/python3.6/urllib/request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.6/urllib/request.py", line 526, in open
    response = self._open(req, data)
  File "/usr/lib/python3.6/urllib/request.py", line 544, in _open
    '_open', req)
  File "/usr/lib/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.6/urllib/request.py", line 1346, in http_open
    return self.do_open(http.client.HTTPConnection, req)
  File "/usr/lib/python3.6/urllib/request.py", line 1321, in do_open
    r = h.getresponse()
  File "/usr/lib/python3.6/http/client.py", line 1331, in getresponse
    response.begin()
  File "/usr/lib/python3.6/http/client.py", line 297, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.6/http/client.py", line 258, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/lib/python3.6/socket.py", line 586, in readinto
    return self._sock.recv_into(b)
socket.timeout: timed out
```

**Comment by yangl326-Dylan** [comment]
mark 这个问题有进展了吗  我的vid的格式也都是错的

**Comment by drpo** [comment]
@denobisipsis the key seems outdated. cc @zhangn1985

**Comment by zhangn1985** [comment]
yes, outdated!

**Comment by denobisipsis** [comment]
Hi,

key is included in cloud.swf. You can read headers from 

http://aplay-vod.cn-beijing.aliyuncs.com/acfun/player.swf

to get the actual player. Take the date 2017030221 and cloud is

http://uvod-res-oyk-acdn.youku.com/player/$date/core/cloud.swf

cloud.swf is lzma compressed (ZWS) and relevant data is inside another ZWS flash. Use rab to decompress cloud.swf, extract internal swf and generate abc classes to search inside the key.

Actual is 2da3ca9e

**Comment by drpo** [comment]
https://gist.github.com/drpo/f2d44514d89d5777f8e0e1c3e41b2218

**Comment by soimort** [comment]
Closed via #2010.

**Comment by soimort** [comment]
Closed via #2010.
